### PR TITLE
[Dependencies] Add readme to /requirements and unpin requirements without comments

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
As discussed in the dependency pod meeting this week, we are unpinning requirements that are not intentionally pinned (with supported comment). Upgrading requirements can be done with `make upgrade-requirements`

### Safety story
No production level requirements were changed.

### Automated test coverage
Yes

### QA Plan
Not needed

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: requirements/dev-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.6
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (millerdev) [inline requirements/dev-requirements.txt:2]: `make requirements` needs to be run with Python 3.6 until all environments are running 3.9.

Comment (biyeun) [reply requirements/dev-requirements.txt]: done

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

File: requirements/prod-requirements.txt [hunk 2]
```diff
@@ -405,7 +405,7 @@ requests==2.25.1
     #   twilio
 requests-oauthlib==1.3.0
     # via -r base-requirements.in
-requests_toolbelt==0.9.1
+requests-toolbelt==0.9.1
```
Comment (dannyroberts) [inline requirements/prod-requirements.txt:408]: I assume these changes to the txt files are just aliases for each other so it's not a real change. Curious if you have more you can share about how that change happened, i.e. whether it was due to a change on our side / within this PR, or a change to the metadata on pypi for these packages or something.

**Review by dannyroberts — APPROVED** [review]

File: requirements/prod-requirements.txt (hunk 2 above)
Comment (biyeun) [reply requirements/prod-requirements.txt:408]: I actually am not sure how this change happened. Even reverting it manually and re-running on `python 3.6.8` produced the same diff, so I just went with it

**Pull request merged**